src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -481,6 +481,12 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
     var value;
 
     if (__DEV__) {
+      warning(
+        !(workInProgress.type.prototype &&
```

Comment:
This is a bit hard to follow. Let's just wrap it in an `if` block with the real condition, and then pass `false` to the `warning` call.